torch/utils/_sympy/reference.py

```diff
@@ -71,7 +71,7 @@ def square(x):
     @staticmethod
     def mod(x, y):
         ret = abs(x) % abs(y)
-        if x < 0:
+        if isinstance(x, (int, sympy.Number)) and x < 0:
```

Comment:
add comment

Reply:
For reference, please cc me on changes like this on future reviews. In this case, it is fine because there is an ongoing refactor that is going to eventually fix this, but in isolation blasting it this way is just bandaging a problem but making the system even more inconsistent.

Reply:
@ezyang I see, sorry about that. Just to understand, what would be the broader fix for this?

Reply:
Well, the problem is that C and Python different semantics for modulus. So the fix involves carefully distinguishing which you are dealing with. It is not, as is done here, to randomly negate the result or not depending on if you have an fx proxy or not